src/query/storage/m3/storage.go

```diff
@@ -434,15 +451,31 @@ func (s *m3storage) CompleteTags(
 	for _, namespace := range namespaces {
 		namespace := namespace // Capture var
 		go func() {
-			defer wg.Done()
+			_, span, sampled := xcontext.StartSampledTraceSpan(ctx,
```

Comment:
Just double checking that this is threadsafe

Reply:
Yeah I believe it is because it is not mutating anything, but rather returning a new context that wraps the pass-in one.